src/t2iapi/device/device_requests.proto

```diff
@@ -42,3 +42,13 @@ Request a list of removable descriptors of the given class.
 message GetRemovableDescriptorsOfClassRequest {
     DescriptorClass descriptor_class = 1;
 }
+
+/*
+Request to set the device's current electrical power source to the provided electrical power source for the component
+represented by a pm:BatteryDescriptor with the provided @Handle.
```

Comment:
Since "ElectricalPowerSource" refers to the device as a whole and not just a component, it is unclear what "the provided electrical power source for the component represented by a pm:BatteryDescriptor with the provided @ Handle" means here, because "ELECTRICAL_POWER_SOURCE_NON_INTERNAL", in case of battery components, refers to all the battery components, not just the one referenced by the given handle.

You have to specify what exactly is supposed to happen e.g. that other components
represented by a pm:BatteryDescriptor are to be turned off and for the given handle the component shall be in use if and only if ELECTRICAL_POWER_SOURCE_INTERNAL has been given.